[Merged by Bors] - chore(Analysis/CStarAlgebra/Basic): assert that the operator norm is not imported

j-loreaux opened the pull request:
In the past we had decided that this file should be relatively lightweight and that pulling in the operator norm required too many dependencies.

---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include at least one commit authored by each
co-author among the commits in the pull request. If necessary, you may 
create empty commits to indicate co-authorship, using commands like so:

git commit --author="Author Name <author@email.com>" --allow-empty -m "add Author Name as coauthor"

When merging, all the commits will be squashed into a single commit listing all co-authors.

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

j-loreaux committed:
chore(Analysis/CStarAlgebra/Basic): avoid importing `Analysis.Norm.Operator.Basic`.

j-loreaux committed:
fix

eric-wieser commented on Mathlib/Analysis/CStarAlgebra/Basic.lean:
```diff
@@ -5,7 +5,6 @@ Authors: Frédéric Dupuis
 -/
 import Mathlib.Analysis.Normed.Group.Hom
 import Mathlib.Analysis.Normed.Module.Basic
-import Mathlib.Analysis.Normed.Operator.LinearIsometry
```
What does `Operator` refer to in this module name? The file doesn't seem to mention the operator norm...

j-loreaux commented:
> I'm confused here; I can add just the `assert_not_exists` and everything is fine. Why did the other content move?

:facepalm: :facepalm: :facepalm:

I think what happened is that I was tracking down imports in other files and got to this one and saw the `import ...Operator...` and thought, oh that's too heavy, we've had this problem before. Removed the import, things broke. Then I moved them and added the `assert_not_exists`.

Thanks for catching this. I was surprised that we developed the theory of linear isometries without talking about the operator norm, but in hindsight, I suppose it makes sense.

j-loreaux committed:
Revert "fix"

This reverts commit 0b01a9fa4291b90a3f35a02e25ce30a5c330287e.

j-loreaux committed:
Revert "chore(Analysis/CStarAlgebra/Basic): avoid importing `Analysis.Norm.Operator.Basic`."

This reverts commit 7ac7ad75554cc428865b508705115c8e5ac8bd17.

j-loreaux committed:
add `assert_not_exists`

eric-wieser commented:
bors d+

Assuming that no moving is required after all, feel free to merge once CI passes.

j-loreaux committed:
Merge branch 'master' into CStarAlgebra-basic-cut-imports
